model/content_entity.go

```diff
@@ -749,3 +751,44 @@ func (w *ContentEntity) UnmarshalJSON(data []byte) error {
 	err = json.Unmarshal(data, &w.Property)
 	return err
 }
+
+//GenerateID adds a generated id to the payload based on the schema
+func GenerateID(payload []byte, entityFactory EntityFactory) ([]byte, error) {
+	schema := entityFactory.Schema()
```

Comment:
why are you passing in the EntityFactory? the schema is already on the object (FromSchemaAndBuilder)